cmd/internal/find/parser/parseExpression.go

```diff
@@ -35,4 +36,93 @@ func parseExpression(tokens []string) (types.EntryPredicate, error) {
+of entries. By default, this set consists of all traversed entries,
+including the specified path. You can use the mindepth/maxdepth options
+to restrict this set.
+
+Expressions consist of primaries and operators. Primaries yield a predicate
+that's set on a specific property of the entry (e.g. name, path, ctime),
+while the operators combine these "primary predicates" to form more
+powerful, expressive predicates. Use "wash find -h" to view the available
+primaries, and "wash find -h <primary>" to view a detailed description of the
+specified primary, including examples of how to use it.
+
+The rest of this section discusses the operators, their precedence, and
+includes various examples of valid expressions.
+
+OPERATORS:
+The operators are listed in order of decreasing precedence.
+
+( expression )
+    Evaluates to true if the parenthesized expression evaluates to true.
+
+! expression
+-not expression
+    Unary NOT operator. Evaluates to true if the expression is false.
+
+expression -and expression
+expression -a expression
+expression expression
+    Logical AND operator. Evaluates to true if both expressions evaluate
+    to true. Note that the second expression is not evaluated if the
+    first expression is false.
+
+expression -or expression
+expression -o expression
+    Logical OR operator. Evaluates to true if both expressions evaluate to
+    true. Note that the second expression is not evaluated if the first
+    expression is true.
+
+EXAMPLES:
+The following examples are shown as given to the shell. 
+
+find . -name "*c"
+    Print out all entries whose cname ends with a "c"
+
+find . \! -name "*c"
+    Print out all entries whose cname does not end with a "c"
+
+find . -name "*c" -mtime +1h
+find . -name "*c" -a -mtime +1h
+    Print out all entries whose cname ends with a "c" and
+    whose last modification time was more than one hour ago.
+
+find . -name "*c" -o -name "*d"
+    Print out all entries whose cname ends with a "c" or a "d"
+
+find . -path "docker*containers*" \( -name "*c" -o -name "*d" \) -a -mtime -1h
+find . -path "docker*containers*" \( -name "*c" -o -name "*d" \) -mtime -1h
+    Print out all the Docker containers whose cname ends with a "c"
+    or a "d" that were modified within the last hour. Note that
+    without the parentheses, the latter part of the expression would
+    have been parsed as '-name "*c" OR ( -name "*d" AND -mtime -1h)',
+    which is a completely different predicate.
+
+find ec2/instances -m .state.name running -a -m .tags[?] .key termination_date -a .value +0h
+find ec2/instances -m .state.name running -m .tags[?] .key termination_date .value +0h
+    Print out all the running EC2 instances whose termination_date tag expired.
+    Note that the ".key termination_date -a .value +0h" portion of the example
+    is parsed as part of the meta primary's expression, not the top level
+    expression parser. The latter parses the example as "Meta(state.name, ...)
```

Comment:
The last sentence doesn't feel like it helps clarify anything to me. There's no way it could be a valid top-level expression, so saying how that parser "would" parse it sounds wrong.

Reply:
It was meant to be informal, but I can see how that can be confusing. I'll remove it for now.